provider/bigquery/bigquery.go

```diff
@@ -146,24 +146,26 @@ func (c *bigQueryClient) RevokeDatasetAccess(ctx context.Context, d *Dataset, us
 	if err != nil {
 		return err
 	}
-
 	bqRole, err := c.ResolveDatasetRole(role)
 	if err != nil {
 		return err
 	}
-	var removeIndex int
-	for i, a := range metadata.Access {
+
+	isAccessFound := false
+	newAccessEntries := []*bq.AccessEntry{}
```

Comment:
will naming it to `remainingAccessEntries` makes it clearer?

Reply:
Yes! Changed as suggested.